### PR TITLE
setup chunking structure

### PR DESCRIPTION
**Pull request by ChrisRackauckas**

### CODE REVIEW

File: src/compute_jacobian_ad.jl [hunk 1]
```diff
@@ -33,29 +60,28 @@ function ForwardColorJacCache(f,x;
...
-
-    ForwardColorJacCache(t,fx,_dx,p,color)
 end
 
 function forwarddiff_color_jacobian!(J::AbstractMatrix{<:Number},
                 f,
                 x::AbstractArray{<:Number};
                 dx = nothing,
-                color)
+                color = 1:length(x))
```
Comment (matbesancon) [inline src/compute_jacobian_ad.jl]:
```suggestion
                color = eachindex(x))
```

File: src/compute_jacobian_ad.jl [hunk 2]
```diff
@@ -8,20 +8,47 @@ struct ForwardColorJacCache{T,T2,T3,T4,T5}
...
+
+getsize(::Val{N}) where N = N
+getsize(N::Integer) = N
+
+function ForwardColorJacCache(f,x,_chunksize = nothing;
                               dx = nothing,
                               color=1:length(x))
 
-    t = zeros(Dual{typeof(f), eltype(x), maximum(color)},length(x))
+    if _chunksize === nothing
```
Comment (matbesancon) [inline src/compute_jacobian_ad.jl]: move the assignment at beginning of expression? `chunksize = if _chunksize ...`

File: src/compute_jacobian_ad.jl [hunk 3]
```diff
@@ -8,20 +8,47 @@ struct ForwardColorJacCache{T,T2,T3,T4,T5}
...
+
+generate_chunked_partials(x,color,N::Integer) = generate_chunked_partials(x,color,Val(N))
+function generate_chunked_partials(x,color,::Val{N}) where N
+
+    # TODO: should only go up to the chunksize each time, and should
+    # generate p[i] different parts, each with less than the chunksize
+
     partials_array = Array{Bool}(undef, length(x), maximum(color))
     for color_i in 1:maximum(color)
         for i in 1:length(x)
```
Comment (matbesancon) [inline src/compute_jacobian_ad.jl]:
```suggestion
        for i in eachindex(x)
```

File: src/compute_jacobian_ad.jl [hunk 4]
```diff
@@ -8,20 +8,47 @@ struct ForwardColorJacCache{T,T2,T3,T4,T5}
...
+    ForwardColorJacCache(t,fx,_dx,p,color)
+end
+
+generate_chunked_partials(x,color,N::Integer) = generate_chunked_partials(x,color,Val(N))
+function generate_chunked_partials(x,color,::Val{N}) where N
+
+    # TODO: should only go up to the chunksize each time, and should
+    # generate p[i] different parts, each with less than the chunksize
+
     partials_array = Array{Bool}(undef, length(x), maximum(color))
```
Comment (matbesancon) [inline src/compute_jacobian_ad.jl]: Could this be lighter on memory:
```suggestion
    partials_array = BitMatrix{Bool}(undef, length(x), maximum(color))
```

File: src/compute_jacobian_ad.jl [hunk 5]
```diff
@@ -8,23 +8,50 @@ struct ForwardColorJacCache{T,T2,T3,T4,T5}
...
+    ForwardColorJacCache(t,fx,_dx,p,color)
+end
+
+generate_chunked_partials(x,color,N::Integer) = generate_chunked_partials(x,color,Val(N))
+function generate_chunked_partials(x,color,::Val{N}) where N
+
+    # TODO: should only go up to the chunksize each time, and should
+    # generate p[i] different parts, each with less than the chunksize
+
+    partials_array = BitMatrix{Bool}(undef, length(x), maximum(color))
```
Comment (ChrisRackauckas) [inline src/compute_jacobian_ad.jl]:
```suggestion
    partials_array = BitMatrix(undef, length(x), maximum(color))
```

**Pull request merged**